stdlib/public/core/Span/Span.swift

```diff
@@ -0,0 +1,827 @@
+//===--- Span.swift -------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// A Span<Element> represents a span of memory which
+// contains initialized instances of `Element`.
+@_disallowFeatureSuppression(NonescapableTypes)
+@frozen
+@available(SwiftStdlib 9999, *)
+public struct Span<Element: ~Copyable & ~Escapable>
+: ~Escapable, Copyable, BitwiseCopyable {
```

Comment:
Coming from a place of ignorance: it’s surprising to me that an object could both have lifetime constraints and be considered “bitwise copyable”. Are there funny interactions we should be aware of? For instance, how would we need to annotate the eventual MutableRawSpan.store signature to mean it can take any BitwiseCopyable but it’s going to escape it?

Reply:
Using that `store` function will be considered an escape (like setting the `pointee` property of a pointer.)